lib/katello/tasks/upgrades/4.1/fix_invalid_pools.rake

```diff
@@ -0,0 +1,62 @@
+namespace :katello do
+  namespace :upgrades do
+    namespace '4.1' do
+      desc "Removed orphaned pools and correct org+subscription mismatch"
+      task :fix_invalid_pools => ["environment", "check_ping"] do
+        logger = Logger.new(STDOUT)
+        invalid_pools = Katello::Pool.all.select(&:invalid?)
+
+        # Make sure Subscriptions are up to date for any org that has invalid pools
+        org_ids = invalid_pools.map(&:organization_id).uniq.compact
```

Comment:
SQL is always faster than Ruby :) 
```suggestion
        invalid_pools = Katello::Pool.all.select(&:invalid?).distinct

        # Make sure Subscriptions are up to date for any org that has invalid pools
        org_ids = invalid_pools.map(&:organization_id).compact
```

Reply:
That won't work here since the .select returns an Array 'cause the validations happen in Ruby land and not the db

Reply:
:cry:

Reply:
hopefully you 'd not encounter too many invalid pools over time. Not a biggie imo

Reply:
Plus this rake task is a one-off as part of the upgrade process. No more invalid pools will be created after this point, and eventually this task will go away